coderd/database/dbpurge/dbpurge_test.go

```diff
@@ -183,20 +184,28 @@ func TestDeleteOldWorkspaceAgentLogs(t *testing.T) {
 		// given
 		agent := mustCreateAgentWithLogs(ctx, t, db, user, org, tmpl, tv, now.Add(-8*24*time.Hour), t.Name())
 
+		// Make sure that agent logs have been collected.
+		agentLogs, err := db.GetWorkspaceAgentLogsAfter(ctx, database.GetWorkspaceAgentLogsAfterParams{
+			AgentID: agent,
+		})
+		require.NoError(t, err)
+		require.NotZero(t, agentLogs, "agent logs must be present")
+
 		// when
 		closer := dbpurge.New(ctx, logger, db)
 		defer closer.Close()
 
 		// then
-		require.Eventually(t, func() bool {
-			agentLogs, err := db.GetWorkspaceAgentLogsAfter(ctx, database.GetWorkspaceAgentLogsAfterParams{
+		assert.Eventually(t, func() bool {
+			agentLogs, err = db.GetWorkspaceAgentLogsAfter(ctx, database.GetWorkspaceAgentLogsAfterParams{
 				AgentID: agent,
 			})
 			if err != nil {
-				return false
+				require.FailNow(t, "unable to get agent logs: %w", err)
```

Comment:
Using `require` outside of the main goroutine can have some negative side-effects:
https://github.com/Antonboom/testifylint#go-require

> ... functions leading to t.FailNow (essentially to runtime.GoExit) must only be used in the goroutine that runs the test. Otherwise, they will not work as declared, namely, finish the test function.

Reply:
Switched to `t.Fatalf` 👍

Reply:
I believe the advice from that link is that you should _not_ use `t.Fatalf` outside the main goroutine.

Reply:
I apologize, I didn't read the note carefully. Refactored to `if err != nil { return false }`, but checking the last value of `err`.